tests/by-util/test_ls.rs

```diff
@@ -6311,3 +6311,336 @@ fn test_ls_time_sort_without_long() {
+            .arg("-1")
+            .arg("--color=never")
+            .succeeds();
+
+        let lines: Vec<&str> = result.stdout_str().lines().collect();
+
+        // In C locale, ASCII comes first, then non-ASCII by byte value
+        // 'a' and 'z' should come before UTF-8 encoded characters
+        assert_in_order(&lines, &["apple", "zebra"]);
+
+        // UTF-8 characters should come after ASCII
+        let zebra_pos = lines.iter().position(|&l| l == "zebra").unwrap();
+        let apfel_pos = lines.iter().position(|&l| l == "äpfel");
+        let eclair_pos = lines.iter().position(|&l| l == "éclair");
+
+        if let (Some(apfel), Some(eclair)) = (apfel_pos, eclair_pos) {
+            assert!(
+                zebra_pos < apfel,
+                "ASCII 'zebra' should come before 'äpfel'"
+            );
+            assert!(
+                zebra_pos < eclair,
+                "ASCII 'zebra' should come before 'éclair'"
+            );
+        }
+    }
+
+    #[test]
+    fn test_ls_locale_german_umlauts() {
+        let locale = "de_DE.UTF-8";
+        if !have_locale(locale) {
+            eprintln!("Skipping test: locale {locale} not available");
+            return;
+        }
+
+        let scene = TestScenario::new(util_name!());
+        let at = &scene.fixtures;
+
+        // Create files with German umlauts
+        at.touch("apfel"); // apple
+        at.touch("äpfel"); // apples (with umlaut)
+        at.touch("bär"); // bear
+        at.touch("öffnung"); // opening
+        at.touch("über"); // over
+        at.touch("zebra");
+
+        let result = scene
+            .ucmd()
+            .env("LC_ALL", locale)
+            .arg("-1")
+            .arg("--color=never")
+            .succeeds();
+
+        let lines: Vec<&str> = result.stdout_str().lines().collect();
+
+        // In German locale, umlauts should sort near their base letters
+        // ä near a, ö near o, ü near u
+        assert_in_order(&lines, &["apfel", "äpfel", "bär"]);
+    }
+
+    #[test]
+    fn test_ls_locale_french_accents() {
+        let locale = "fr_FR.UTF-8";
+        if !have_locale(locale) {
+            eprintln!("Skipping test: locale {locale} not available");
+            return;
+        }
+
+        let scene = TestScenario::new(util_name!());
+        let at = &scene.fixtures;
+
+        // Create files with French accents
+        at.touch("ecole"); // school
+        at.touch("école"); // school (with accent)
+        at.touch("etude"); // study
+        at.touch("étude"); // study (with accent)
+        at.touch("zebra");
+
+        let result = scene
+            .ucmd()
+            .env("LC_ALL", locale)
+            .arg("-1")
+            .arg("--color=never")
+            .succeeds();
+
+        let lines: Vec<&str> = result.stdout_str().lines().collect();
+
+        // Accented letters should sort near their base letters
+        assert_in_order(&lines, &["ecole", "école", "etude", "étude"]);
```

Comment:
Could you also check that the order is not the same with the `C` or `en_US` locale, for example?
(same for the other tests)